lib/facter/util/facts/posix/virtual_detector.rb

```diff
@@ -4,79 +4,67 @@ module Facter
 
-            @@fact_value
-          end
+              @@fact_value
+            end
 
-          private
+            private
 
-          def check_docker_lxc
-            @log.debug('Checking Docker and LXC')
-            Facter::Resolvers::Containers.resolve(:vm)
-          end
+            def check_docker_lxc
+              Facter::Resolvers::Containers.resolve(:vm)
+            end
 
-          def check_gce
-            @log.debug('Checking GCE')
-            bios_vendor = Facter::Resolvers::Linux::DmiBios.resolve(:bios_vendor)
-            'gce' if bios_vendor&.include?('Google')
-          end
+            def check_gce
+              bios_vendor = Facter::Resolvers::Linux::DmiBios.resolve(:bios_vendor)
+              'gce' if bios_vendor&.include?('Google')
+            end
 
-          def check_vmware
-            @log.debug('Checking VMware')
-            Facter::Resolvers::Vmware.resolve(:vm)
-          end
+            def check_vmware
+              Facter::Resolvers::Vmware.resolve(:vm)
+            end
 
-          def retrieve_from_virt_what
-            @log.debug('Checking virtual_what')
-            Facter::Resolvers::VirtWhat.resolve(:vm)
-          end
+            def retrieve_from_virt_what
+              Facter::Resolvers::VirtWhat.resolve(:vm)
+            end
 
-          def check_open_vz
-            @log.debug('Checking OpenVZ')
-            Facter::Resolvers::OpenVz.resolve(:vm)
-          end
+            def check_open_vz
+              Facter::Resolvers::OpenVz.resolve(:vm)
+            end
 
-          def check_vserver
-            @log.debug('Checking VServer')
-            Facter::Resolvers::VirtWhat.resolve(:vserver)
-          end
+            def check_vserver
+              Facter::Resolvers::VirtWhat.resolve(:vserver)
+            end
 
-          def check_xen
-            @log.debug('Checking XEN')
-            Facter::Resolvers::Xen.resolve(:vm)
-          end
+            def check_xen
+              Facter::Resolvers::Xen.resolve(:vm)
+            end
 
-          def check_freebsd
-            return unless Object.const_defined?('Facter::Resolvers::Freebsd::Virtual')
+            def check_freebsd
+              return unless Object.const_defined?('Facter::Resolvers::Freebsd::Virtual')
 
-            @log.debug('Checking if jailed')
-            Facter::Resolvers::Freebsd::Virtual.resolve(:vm)
-          end
+              Facter::Resolvers::Freebsd::Virtual.resolve(:vm)
+            end
 
-          def check_other_facts
-            @log.debug('Checking others')
-            product_name = Facter::Resolvers::Linux::DmiBios.resolve(:product_name)
-            bios_vendor =  Facter::Resolvers::Linux::DmiBios.resolve(:bios_vendor)
-            return 'kvm' if bios_vendor&.include?('Amazon EC2')
-            return unless product_name
+            def check_other_facts
+              product_name = Facter::Resolvers::Linux::DmiBios.resolve(:product_name)
```

Comment:
Not related to you changes, but i think this can be improved a bit.
We do not need to resilve product_name if `return 'kvm' if bios_vendor&.include?('Amazon EC2')` can lead to it not being used

I guess would be better to resolve it, if it will be used:

```ruby
bios_vendor =  Facter::Resolvers::Linux::DmiBios.resolve(:bios_vendor)
return 'kvm' if bios_vendor&.include?('Amazon EC2')

product_name = Facter::Resolvers::Linux::DmiBios.resolve(:product_name)
return unless product_name
```